### PR TITLE
add priority class to the daemonset

### PR DESCRIPTION
**Pull request by QuantumEnigmaa**
Towards https://github.com/giantswarm/giantswarm/issues/30815

### CODE REVIEW

File: helm/fluent-logshipping-app/templates/priorityClass.yaml
Comment (QuentinBisson) [inline helm/fluent-logshipping-app/templates/priorityClass.yaml:10]: I think it's too high :D

File: helm/fluent-logshipping-app/templates/priorityClass.yaml
Comment (QuentinBisson) [inline helm/fluent-logshipping-app/templates/priorityClass.yaml:1]: customEnabled? Can you maybe use something like 
```suggestion
{{- if .Values.fluentbit.priorityClass.create }}
```

File: helm/fluent-logshipping-app/templates/priorityClass.yaml
Comment (QuentinBisson) [inline helm/fluent-logshipping-app/templates/priorityClass.yaml:10]: Also, maybe we can override this one so it should be part of the values?

File: helm/fluent-logshipping-app/values.yaml
Comment (QuentinBisson) [inline helm/fluent-logshipping-app/values.yaml:12]:
```suggestion
    create: true
```

File: helm/fluent-logshipping-app/templates/priorityClass.yaml
Comment (QuentinBisson) [reply helm/fluent-logshipping-app/templates/priorityClass.yaml:10]: Oh no it's fine :)

File: helm/fluent-logshipping-app/templates/priorityClass.yaml
Comment (QuantumEnigmaa) [reply helm/fluent-logshipping-app/templates/priorityClass.yaml:1]: Sure, makes more sense

File: helm/fluent-logshipping-app/templates/priorityClass.yaml
Comment (QuantumEnigmaa) [reply helm/fluent-logshipping-app/templates/priorityClass.yaml:10]: Sure.

Comment (QuantumEnigmaa) [reply helm/fluent-logshipping-app/templates/priorityClass.yaml:10]: Well it's still lower than prometheus which already has the lowest priority among those deployed :shrug:

Comment (QuentinBisson) [reply helm/fluent-logshipping-app/templates/priorityClass.yaml:10]: Yes I was wrong with the number of zeros :D

Comment (QuentinBisson) [reply helm/fluent-logshipping-app/templates/priorityClass.yaml:10]: I thought Prometheus was 500_000 not 500_000_000

File: helm/fluent-logshipping-app/values.yaml
Comment (QuentinBisson) [inline helm/fluent-logshipping-app/values.yaml:13]:
```suggestion
    # -- name of the priority class, leave it empty if you do not want to specify a priority class
    name: fluentbit
```

**Comment by QuentinBisson** [comment]
Apart from my 2 comments it looks good if that was tested on a MC :)

**Comment by QuentinBisson** [comment]
Supersedes https://github.com/giantswarm/fluent-logshipping-app/pull/178

File: helm/fluent-logshipping-app/templates/daemonset.yaml
Comment (QuentinBisson) [reply helm/fluent-logshipping-app/templates/daemonset.yaml:22]: @QuantumEnigmaa what do you think about this one?

File: helm/fluent-logshipping-app/templates/priorityClass.yaml
Comment (QuantumEnigmaa) [reply helm/fluent-logshipping-app/templates/priorityClass.yaml]: Oh ok x)

**Review by QuentinBisson — APPROVED** [review]

**Comment by QuantumEnigmaa** [comment]
Tested and working on `alpaca` : 
```
zirco@Zirko-laptop:~/fluent-logshipping-app$ k get pods -A | grep fluent
monitoring          fluent-logshipping-app-6qbnp                                        1/1     Running            0                 4m49s
monitoring          fluent-logshipping-app-dqftv                                        1/1     Running            0                 5m18s
monitoring          fluent-logshipping-app-dwrfx                                        1/1     Running            0                 5m32s
monitoring          fluent-logshipping-app-gg4k2                                        1/1     Running            0                 4m26s
monitoring          fluent-logshipping-app-gg62q                                        1/1     Running            0                 5m4s
monitoring          fluent-logshipping-app-h2n5h                                        1/1     Running            0                 4m12s
monitoring          fluent-logshipping-app-lzsps                                        1/1     Running            0                 3m46s
zirco@Zirko-laptop:~/fluent-logshipping-app$ k get priorityclasses.scheduling.k8s.io fluentbit 
NAME        VALUE       GLOBAL-DEFAULT   AGE
fluentbit   200000000   false            5m57s
```

**Comment by QuantumEnigmaa** [comment]
I'll wait for additional reviews before merging :)

File: helm/fluent-logshipping-app/templates/daemonset.yaml
Comment (QuentinBisson) [reply helm/fluent-logshipping-app/templates/daemonset.yaml:22]: Those are some valid points yes. Maybe we want to have it disabled and enable it in shared configs

Comment (QuantumEnigmaa) [reply helm/fluent-logshipping-app/templates/daemonset.yaml:22]: Works for me. Gonna add this here

Comment (QuantumEnigmaa) [reply helm/fluent-logshipping-app/templates/daemonset.yaml]: Done.

**Review by hervenicol — APPROVED** [review]
LGTM :+1:

**Pull request merged**
